libraries/botframework-schema/src/activityInterfaces.ts

```diff
@@ -300,3 +300,61 @@ export interface ITraceActivity extends IActivity {
 export interface IHandoffActivity extends IActivity {
 }
 
+/**
+ * The value field of a ICommandActivity contains metadata related to a command.
+ * An optional extensible data payload may be included if defined by the command activity name.
+ */
+export interface CommandValue<T> {
+    /**
+     * ID of the command.
+     */
+    commandId: string;
+    /**
+     * The data field containing optional parameters specific to this command activity,
+     * as defined by the name. The value of the data field is a complex type.
+     */
+    data?: T;
+}
+
+/**
+ * Asynchronous external command.
+ */
+export interface ICommandActivity<T> extends IActivity {
+    /**
+     * The name of the event.
+     */
+    name: string;
+    /**
+     * The value for this command.
+     */
+    value?: CommandValue<T>;
+}
```

Comment:
Note: I wasn't sure whether it was better to go with type generics here or something like `Record<string, unknown>`. Let me know if you want it changed.

Reply:
This makes sense - I wonder if there's some implicit meaning behind the fact that the .NET fields are JSON properties. This is a bit pedantic, but we can be even more strict with `T`. For example,

```typescript
type Foo<T extends number | string | boolean> = Promise<T>;
```

Informs the compiler that `T` must "be assignable to" one of the types in the union of `number | string | boolean`.

There's a good chance I'm over-engineering here, but it's something to consider even if it's just for the sake of learning about generics in Typescript 😄

Reply:
Would it make sense to have ICommandActivity without T, and the value as any, then let consumers type it as they retrieve the .value?